### ISSUE TITLE
Could you release the details of the dataset splitting?

### ISSUE THREAD

**Issue by yinheyanxian** [ifs-rwth-aachen/GERALD#3]
I want to evaluate the performance of the algorithm under different lighting and weather conditions as in the paper, but my custom splitting may not necessarily include this data well. Could you release the details of the dataset splitting? Just like in the paper, 80% is used for training and 20% for validation, which images are used for training and which images are used for validation?

![1736047288351](https://github.com/user-attachments/assets/2e3eb0a5-1e1a-4122-bf35-39039fcf4863)

**Comment by philipp-leibner** [comment]
Hi,

i wrote the code a couple of year ago and havent really touched it since, but i think if you use something like the following:
```python
files = sorted(os.listdir(self.an_path))
filenames = [os.path.splitext(filename)[0] for filename in files]
```

with 
```python
test_idxs = random.choices(np.arange(0, len(filenames), 1), k=n_test_images)
```
and 
```python
random.seed(331297)
```
you should likely get the same train/test split.

**Comment by yinheyanxian** [comment]
> Hi,
> 
> i wrote the code a couple of year ago and havent really touched it since, but i think if you use something like the following:
> 
> ```python
> files = sorted(os.listdir(self.an_path))
> filenames = [os.path.splitext(filename)[0] for filename in files]
> ```
> 
> with
> 
> ```python
> test_idxs = random.choices(np.arange(0, len(filenames), 1), k=n_test_images)
> ```
> 
> and
> 
> ```python
> random.seed(331297)
> ```
> 
> you should likely get the same train/test split.

Thanks.
So your data splitting is random, right?
I have calculated the images for each weather conditions based on additional JSON files, and divided the training and validation sets in an 8:2 ratio for each weather category. Then, they were integrated together. Finally, the mAP of the model under each weather condition is evaluated. Do you think this is reasonable?

**Comment by philipp-leibner** [comment]
Yes should be reasonable. I also notices that i shuffled all filenames immediately after discovery and then applied the code mentioned above.

**Comment by yinheyanxian** [comment]
> Yes should be reasonable. I also notices that i shuffled all filenames immediately after discovery and then applied the code mentioned above.

Thank you.
